pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -645,6 +660,7 @@ async def get_instructions(run_context: RunContext[AgentDepsT]) -> str | None:
             },
         )
 
+        run_metadata: str | dict[str, str] | None = None
```

Comment:
We could technically set a non-callable metadata even if the run fails, and the ctx isn't available. What do you think? I opted for the simpler implementation.